pyro/distributions/avf_mvn.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+from torch.distributions import constraints
+
+from pyro.distributions.torch import MultivariateNormal
+from pyro.distributions.util import sum_leftmost
+
+
+class AVFMultivariateNormal(MultivariateNormal):
+    """Multivariate normal (Gaussian) distribution with transport equation inspired control
+    variates (adaptive velocity fields).
+
+    A distribution over vectors in which all the elements have a joint Gaussian density.
+
+    :param torch.Tensor loc: D-dimensional mean vector.
+    :param torch.Tensor scale_tril: Cholesky of Covariance matrix; D x D matrix.
+    :param torch.Tensor control_var: 2 x L x D tensor that parameterizes the control variate;
+    L is an arbitrary positive integer.
+    This parameter needs to be learned (i.e. adapted) to achieve lower variance gradients.
+
+    Example usage::
+
+    control_var = torch.tensor(0.1 * torch.ones(2, 1, D), requires_grad=True)
+    opt_cv = pyro.optim.Adam([control_var], lr=0.1, betas=(0.5, 0.999))
+
+    for _ in range(1000):
+        d = AVFMultivariateNormal(loc, scale_tril, control_var)
+        z = d.rsample()
+        cost = torch.pow(z, 2.0).sum()
+        cost.backward()
+        opt_cv.step()
+        opt_cv.zero_grad()
+
+    """
+    arg_constraints = {"loc": constraints.real, "scale_tril": constraints.lower_triangular,
+                       "control_var": constraints.real}
+
+    def __init__(self, loc, scale_tril, control_var):
+        assert(loc.dim() == 1), "AVFMultivariateNormal loc must be 1-dimensional"
```

Comment:
Use `if ...: raise ValueError(...)` for user facing error messages. `assert` should be for developer-facing errors.

Reply:
done